app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -257,7 +257,7 @@ private void doMenuAction(int menuItemId){
                 Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
                 mDrawerLayout.closeDrawers();
                 startActivity(intent);
-
+				finish();
```

Comment:
This is not a clean solution, `closeDrawers()` is an asynchronous operation. You should wait for the drawers to close before starting an activity.